library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -442,9 +443,16 @@ private Slice readOrSkip(boolean skip) throws IOException
           inputStream = getInputStream(currentPointer);
         }
 
-        if (inputStream != null && currentPointer.offset <
-            fileSystemWAL.fileContext.getFileStatus(currentOpenPath).getLen()) {
-          int len = inputStream.readInt();
+        int ch1;
+
+        if (inputStream != null && (ch1 = inputStream.read()) != -1) {
```

Comment:
with readInt(), reading of every file ends with exception, that is not the case with read().